app/src/main/java/io/rapidpro/androidchannel/IncomingSMSObserver.java

```diff
@@ -112,4 +123,21 @@ public void onChange(boolean selfChange) {
             RapidPro.get().sync(true);
         }
     }
+
+    private boolean matchBlacklistedNumbers(String blacklistedNumbers, String address) {
+        if (!TextUtils.isEmpty(blacklistedNumbers)) {
+            String [] numbers = blacklistedNumbers.split(",");
+            for (String number : numbers) {
+                if (address.toLowerCase().contains(number.trim().toLowerCase())) {
```

Comment:
May want to check that number is non-empty here (after trimming), if someone has a trailing comma then it would match all numbers.